### ISSUE TITLE
Request to add support for WebSocket

### ISSUE THREAD

**Issue by thevops** [prometheus/blackbox_exporter#1081]
There is an old issue about WebSocket probe: https://github.com/prometheus/blackbox_exporter/issues/536

It'd be nice to reconsider adding WebSocket support to Blackbox. The popularity of this protocol is growing.
What do you think?

**Comment by SuperQ** [comment]
Seems like a good idea. Can you write out a proposal for what this would look like? What additional parameters / features would be needed.

**Comment by thevops** [comment]
# Proposal

The popularity of the WebSocket protocol is growing. Having support for this protocol would be very useful.

## Features

The WebSocket probe could have the same attributes as the HTTP probe, like support for Basic Auth or TLS checking. Although a Minimum Viable Product can support only essential functions: sending requests with query data, receiving answers, and comparing received data to expected data.

## Feedback

I’m ready to test the WebSocket probe implementation, provide feedback and contribute as much as I can.

**Comment by muety** [comment]
Any changes to have blackbox exporter support websocket any time soon?

**Comment by adnull** [comment]
@muety as the PR is still not accepted you can try using websocket support version directly from my repo. 
https://github.com/adnull/blackbox_exporter/tree/websocket-support

**Comment by electron0zero** [comment]
left some review comments on https://github.com/prometheus/blackbox_exporter/pull/1278#pullrequestreview-3397863502, will review it again once comments are addressed.

**Comment by ivankatliarchuk** [comment]
I think this one is resolved https://github.com/prometheus/blackbox_exporter/pull/1278#pullrequestreview-3397863502. Maybe worth to close or something else left?

**Comment by electron0zero** [comment]
We are missing `probe_websocket_ssl` metric for for websocket prober: @adnull said they will send a follow up PR to add it https://github.com/prometheus/blackbox_exporter/pull/1278#discussion_r2757984786

so I am looking for PR to add probe_websocket_ssl and then websocket will be in paririty with other probers.

**Comment by adnull** [comment]
Yeah, I'm still lack of time to manage that. But still hoping to implement it.

**Comment by adnull** [comment]
@electron0zero added the metric and unified SSL metrics registry in the https://github.com/prometheus/blackbox_exporter/pull/1574 PR
